tests/integration/test_schema.py

```diff
@@ -1096,30 +1094,42 @@ async def test_schema_types(registry_async_client, trail):
 
 
 @pytest.mark.parametrize("trail", ["", "/"])
-async def test_schema(registry_async_client, trail):
-    subject = new_random_name("subject")
-    schema_str = '{"type": "string"}'
+async def test_schema_repost(registry_async_client, trail):
+    """"
+    Repost same schema again to see that a new id is not generated but an old one is given back
+    """
+    subject = create_subject_name_factory(f"test_schema_repost-{trail}")()
+    unique_field_factory = create_field_name_factory(trail)
+
+    unique = unique_field_factory()
+    schema_str = '{"type": "string", "unique" : "%s"}' % unique
```

Comment:
```suggestion
    schema_str = json.dumps({"type": "string", "unique" : unique})
```
Using a serializer is less error prone than writing JSON by hand. The code can also benefit of the linting, and formatting by the static tools, and gets easier to read because one doesn't have to escape double quotes.

This also applies to the other tests.

Reply:
Yes.... In #222 I've been changing these to use `jsonlib.dumps`. Especially when checking against against Schema Registry the format of the incoming JSON string is something that we don't know. Also, this PR doesn't introduce this practise of checking equality of two JSONs using strings - it was already there.

The scope of this PR is to split the code, not try to make everything perfect. There's still a lot of "non-optimal" stuff in the code.

I wouldn't do these fixes here. They'll be fixed to large extent in #222.